### ISSUE TITLE
Re-processing does not mark jobs as BLOCKED

### ISSUE THREAD

**Issue by dfaller** [ngageoint/scale#711]
There is a bug in re-processing where new jobs are not marked as BLOCKED when they should.

For instance consider a recipe R1: A1 (FAILED) -> B1 (BLOCKED) -> C1 (BLOCKED)
If you do a re-process of the recipe affecting only jobs B and C, you will get the following new recipe:
R2: A1 (FAILED) -> B2 (PENDING) -> C2 (PENDING)
when you should get this:
R2: A1 (FAILED) -> B2 (BLOCKED) -> C2 (BLOCKED)

Basically the re-processing code that creates new jobs does not check the recipe to see if those new jobs need to be set to BLOCKED.
